### PR TITLE
Request-signing types + doc skeleton

### PR DESCRIPTION
**Pull request by olorin**
@markhibberd just a start, does this look like it's going in the right direction?

Would be good to whiteboard this at some point so I'm clear on what the functionality needs to be.

### CODE REVIEW

File: doc/request-signing.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+FIXME: finish this section/specify the rest properly
+
+## Signed data
+
+The following data is included in the request's signature:
+
+ - algorithm
+ - request date
+ - request scope
```
Comment (markhibberd) [inline doc/request-signing.md]: What does request scope mean here? Is this just the key id or does it include the key id? If not the id needs to be in here somewhere.

Comment (olorin) [inline doc/request-signing.md]: I'm not sure yet, was one of the things I was hoping to talk to you about. Was thinking something like `API/service + key_id`.

**Comment by markhibberd** [comment]
Yep, this looks pretty close to what I imagined. Whiteboard would be good, Friday? But for the time being this looks good. The one thing that I didn't click to before you wrote that down is that it has to be remarkably similar to the AWS V4 algorithm, I haven't compared notes, just from memory, I didn't really click to how general it was. The question then becomes can/should we try to make it identical (with different base key-id generation), not sure if there is an advantage either way just a thought.

👍

File: doc/request-signing.md (hunk 1 above)
Comment (markhibberd) [inline doc/request-signing.md]: Ok, makes sense I guess, the API/service key is also going to be locked in from the signing of the URI path, but being explicit about this doesn't hurt.

Comment (markhibberd) [inline doc/request-signing.md]: (even if we keep in fairly coarse for now and only had the one "service" exposed)

**Comment by olorin** [comment]
Friday sounds good.

> The question then becomes can/should we try to make it identical (with different base key-id generation), not sure if there is an advantage either way just a thought.

The version I have in mind is based on a partially-complete spec called [escher](https://escherauth.io/) which I wasn't intending to implement strictly, just use as a baseline - it's in turn based on AWSV4. I'm not sure wrt making it identical; a few of our requirements seem a little different, e.g., I'm not sure if we want to restrict scope by region or not. Then again, it could be convenient on the client side? I'm not sure how many external API clients we're expecting though.

**Pull request merged**